File: frontend/src/objects/message/response/MessageInboxResponse.ts
import type { MessageConversationSummary } from '@/objects/message/response/MessageConversationSummary'

/** 私信收件箱响应；包含会话分页和总未读数。 */
export type MessageInboxResponse = {
  conversations: MessageConversationSummary[]
  totalUnreadCount: number
  page: number
  pageSize: number
  totalItems: number
}

File: backend/src/main/scala/domains/message/objects/response/MessageInboxResponse.scala
package domains.message.objects.response

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

/** 私信收件箱分页响应，包含会话列表、总未读数和分页元数据。 */
final case class MessageInboxResponse(
  conversations: List[MessageConversationSummary],
  totalUnreadCount: Int,
  page: Int,
  pageSize: Int,
  totalItems: Long
)

/** 提供私信收件箱响应 JSON codec。 */
object MessageInboxResponse:
  given Encoder[MessageInboxResponse] = deriveEncoder[MessageInboxResponse]
  given Decoder[MessageInboxResponse] = deriveDecoder[MessageInboxResponse]
